packages/commerce-server/src/record-new-purchase.ts

```diff
@@ -137,7 +137,6 @@ export async function recordNewPurchase(checkoutSessionId: string): Promise<{
         // 1. Create bulk coupon record
         const coupon = await transaction.coupon.create({
           data: {
-            bulkPurchaseId: purchase.id,
```

Comment:
This discontinues the writes, the changes in the other file remove the one place where this value was being read.